scipy/stats/_qmc.py

```diff
@@ -749,7 +749,7 @@ def __init__(
 
         self.d = d
         self.rng = check_random_state(seed)
-        self.rng_seed = copy.deepcopy(seed)
+        self.rng_seed = copy.deepcopy(self.rng)
```

Comment:
This handles the case when `seed=None` okay, but doesn't fix anything when `seed` is an existing `Generator` instance. It's not safe to `deepcopy()` a `Generator` instance that you did not create. You will need to spawn in that case.

Reply:
Actually spawning (which is done in the `_rng_spawn()` but not the code snippet posted here) will advance a counter in the underlying `SeedSequence`. That's all that's needed to ensure that multiple instantiations of a `QMCEngine` with the same `Generator` instance will get distinct PRNG states.

I would spawn if and only if we get a `Generator` rather than a static seed or `None`.

```python
if isinstance(seed, Generator):
    # Spawn a Generator that we can own and reset.
    self.rng = _rng_spawn(seed, 1)[0]
else:
    self.rng = check_random_state(seed)
self.rng_seed = copy.deepcopy(self.rng)
```

There's a lingering issue with passing in a live `RandomState` instance. Because those are almost always created from legacy `MT19937` seeding rather than `SeedSequence`, there's nothing that can be spawned. I'd probably just document the issue, maybe with a `warn()` there if you like.

Reply:
oh perfect then 👍 I will add that snippet and a comment as you suggest.

Reply:
What do you think if instead we moved the spawn logic into `check_random_state` (the one in QMC, not util)? Seems like this should always apply no?

Reply:
No, the use in `van_der_corput()` should remain as-is. These semantics are very specific to the one usage in `QMCEngine.__init__()`, so it makes sense to keep that logic inlined there. Obviously, you can refactor it however you like, but if you do factor it out into its own function, I do suggest not naming it `check_random_state()` to avoid confusion. The current version of QMC's `check_random_state()` is still along the lines of where we want the `_util` `check_random_state()` to go, eventually, but the spawning version would not be.